repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/atomfeed/FeedCacheInvalidator.java

```diff
@@ -65,7 +66,13 @@ public void run() {
 
             // Iterate through atom feeds to retrieve tokens and users to invalidate from Repose cache
             for (AuthFeedReader rdr : feeds) {
-                CacheKeys keys = rdr.getCacheKeys();
+                CacheKeys keys = null;
+                try {
+                    keys = rdr.getCacheKeys();
+                } catch (TimeoutException e) {
+                    LOG.error("Unable get Cached Keys.");
+                    LOG.trace("", e);
+                }
 
                 userKeys.addAll(keys.getUserKeys());
```

Comment:
you just created a potential NPE, if the call ti get cache keys throws a timeout exception, this line will throw an npe.